### PR TITLE
[DX-1652] Plugin Bundle CLI Improvements

### PR DESCRIPTION
**Pull request by dcs3spp**
### For internal users - Please add a Jira DX PR ticket to the subject!
[DX-1652](https://tyktech.atlassian.net/browse/DX-1652)

---

### Preview Link

[preview](https://deploy-preview-5401--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/advanced-configurations/plugins/bundles/bundle-cli)

### Description

Improve bundle CLI content to give required prerequisites following [previous review](https://github.com/TykTechnologies/tyk-docs/pull/5330#discussion_r1742190970)

#### Screenshots (if appropriate)
<br>

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have reviewed the suggestions made by our AI (PR Agent) and updated them accordingly (spelling errors, rephrasing, etc.)
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labeled** the PR

[DX-1652]: https://tyktech.atlassian.net/browse/DX-1652?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

___

### **PR Type**
documentation

___

### **Description**
- Added a new section on prerequisites for creating plugin bundles, detailing necessary files like `manifest.json`, plugin source code files, and optional certificate keys.
- Provided a suggested directory structure for organizing plugin bundles in Golang, Javascript, and Python.
- Included a Docker example for creating plugin bundles for Tyk Gateway versions earlier than 5.5, detailing the command and its purpose.
- Enhanced the documentation with additional command options and notes for creating plugin bundles, improving clarity and usability.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>bundle-cli.md</strong><dd><code>Enhance Plugin Bundle CLI Documentation with Prerequisites and Docker </code><br><code>Example</code></dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/plugins/bundles/bundle-cli.md

<li>Added prerequisites section detailing required files and <br>configurations for plugin bundles.<br> <li> Introduced a directory structure guide for Golang, Javascript, and <br>Python bundles.<br> <li> Added a Docker example for creating plugin bundles for Gateway <br>versions < 5.5.<br> <li> Enhanced documentation with command options and additional notes for <br>bundle creation.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5401/files#diff-5ebd83bb172c004d567c6aa838773003fe4ecfe16a33da0bb559a9894594510f">+86/-5</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/plugins/bundles/bundle-cli.md [hunk 1]
```diff
@@ -13,7 +13,66 @@ Issue the following command to see more details on the `bundle` command:
...
+
+- **Manifest.json**: The [manifest.json]({{< ref "plugins/how-to-serve-plugins/plugin-bundles#manifest" >}}) file contains the paths to the plugin source files and the name of the function implementing each plugin. The *manifest.json* file is mandatory and must exist on the Tyk Gateway file system. By default the bundle file looks for a file named *manifest.json* in the current working directory where the bundle command is run from. The exact location can be specified using the `--manifest` command option.
+- **Plugin source code files**: The plugin source code files should be contained relative to the directory in which the *manifest.json* file is located. The *manifest.json* should contain relative path references to source code files. Please note that source code files are not required when creating a plugin bundle for gRPC plugins since the plugin source code is located at the gRPC server.
+- **Certificate key**: Plugin bundles can optionally be signed with an RSA private key. The corresponding public key should be located in the file configured in environmental variable `TYK_GW_PUBLICKEYPATH` or the `public_key_path` parameter in `tyk.conf`:
+
+```bash
```
Comment (letzya) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/plugins/bundles/bundle-cli.md]:
```suggestion
```json
```

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/plugins/bundles/bundle-cli.md [hunk 2]
```diff
@@ -13,7 +13,66 @@ Issue the following command to see more details on the `bundle` command:
...
+
+---
+
+## Directory Structure
+
+A suggested directory structure is shown below for Golang, Javascript and Python bundles in the tabs below.
+
+In this case, the `manifest.json` will reference the Python files located in the plugins subdirectory, ensuring the Python plugin source files are organized relative to the manifest. The Tyk Gateway will load and execute these Python plugins based on the paths defined in the `manifest.json` file.
+
+{{< tabs_start >}}
```
Comment (letzya) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/plugins/bundles/bundle-cli.md]: We should not add tabs to the docs as they cause bad ux when you search for the content with ctrl+f (it shows as found but you can't see it). We change it to collapse/expand or maybe you have another idea @sharadregoti. Regardless, in this case, it's very short content so I'll just keep it visible

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/plugins/bundles/bundle-cli.md [hunk 3]
```diff
@@ -2,33 +2,110 @@
...
-### Creating a plugin bundle
+---
+
+## Prerequisites
+
+To create plugin bundles you will need the following:
+
+- **Manifest.json**: The [manifest.json]({{< ref "plugins/how-to-serve-plugins/plugin-bundles#manifest" >}}) file
+  contains the paths to the plugin source files and the name of the function implementing each plugin. The
+  _manifest.json_ file is mandatory and must exist on the Tyk Gateway file system. By default the bundle file looks for
```
Comment (sharadregoti) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/plugins/bundles/bundle-cli.md]:
```suggestion
  _manifest.json_ file is mandatory and must exist on the Tyk Gateway file system. By default the bundle CLI looks for
```

**Review by sharadregoti — APPROVED** [review]
Tabs are not working in the live docs, this has to be fixed.

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/plugins/bundles/bundle-cli.md (hunk 2 above)
Comment (sharadregoti) [reply tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/plugins/bundles/bundle-cli.md]: > We should not add tabs to the docs as they cause bad UX

The above statement should only apply for long content, but for short code snippets across languages it is ok to use `Tabs`

I think the use case of `Tabs` here is valid.

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/plugins/bundles/bundle-cli.md [hunk 4]
```diff
@@ -2,33 +2,119 @@
 date: 2024-08-20T13:32:12Z
 title: Bundler CLI Tool
 description: "Explains usage of the bundler CLI tool"
-tags: [ "Tag TODO" ]
+tags: ["Tag TODO"]
```
Comment (sharadregoti) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/plugins/bundles/bundle-cli.md]:
```suggestion
tags: ["Bundle CLI Tool"]
```

**Pull request merged**